tests/specs/examples/erc20-bin-runtime.k.expected

```diff
@@ -0,0 +1,134 @@
+requires "edsl.md"
+
+module ERC20-BIN-RUNTIME
+    imports public EDSL
+    
+    syntax Bool ::= "unknownBool" [klabel(unknownBool), symbol(), no-evaluators(), function(), functional()]
+    
+    syntax ERC20Contract ::= "ERC20" [klabel(contract_ERC20)]
+    
+    syntax ERC20Storage ::= "_balances" "[" Int "]" 
+    
+    syntax ERC20Storage ::= "_allowances" "[" Int "]" "[" Int "]" 
+    
+    syntax ERC20Storage ::= "_totalSupply" 
+    
+    syntax ERC20Storage ::= "_decimals" 
+    
+    syntax ERC20Storage ::= "_name" 
+    
+    syntax ERC20Storage ::= "_symbol" 
+    
+    syntax Int ::= ERC20Contract "." ERC20Storage [klabel(storage_ERC20), macro()]
+    
+    rule  ( ERC20._balances[V0] => #hashedLocation("Solidity", 0 +Int 0, V0) )
+      
+       
+      
+    
+    rule  ( ERC20._allowances[V0][V1] => #hashedLocation("Solidity", #hashedLocation("Solidity", 0 +Int 1, V0), V1) )
```

Comment:
Hmmm, we're not really taking advantage of the already-recursive definition of `#hashedLocation` here...

Any reason you chose to do it this way instead of say:

```suggestion
    rule  ( ERC20._allowances[V0][V1] => #hashedLocation("Solidity", 0 +Int 1, V0 V1 .IntList)
```

Reply:
This is for simplicity of implementation: this way, the structure of the term encoding the base address does not have to be considered when constructing the term encoding an element at a given key.

There a couple of similar simplifications that can be made for a nicer output (these are hinted on in comments). I opened an issue: https://github.com/runtimeverification/erc20-verification/issues/302